src/main/java/org/broadinstitute/hellbender/tools/exome/HashedListExonCollection.java

```diff
@@ -103,27 +110,25 @@ private static void checkForOverlaps(final List<SimpleInterval> sortedIntervals)
         return result;
     }
 
-    /**
-     * Produces the unique name for a genomic interval.
-     *
-     * <p>This name can depend in either the index of the interval, its location or both</p>
-     *
-     * <p>
-     *     The implementation of this method is such that the name will be unique, i.e. that two different locations
-     *     won't ever result in the same name.
-     * </p>
-     *
-     * @param location the genome location of the interval. Is assumed not to be null.
-     *
-     * @return can return {@code null} indicating that this interval should remain anonymous.
-     */
     @Override
-    public String name(final SimpleInterval location) {
-        return location.toString();
+    public String name(final Object exon) {
+        if (exon == null) {
+            throw new IllegalArgumentException("null exon not allowed");
+        } else if (exon instanceof Locatable) {
```

Comment:
When will exon ever not be an instance of `Locatable`? Isn't this more evidence that the parameter of this method should be of type T?